pkg/controllers/readiness/wellknown_ready_controller.go

```diff
@@ -106,9 +111,21 @@ func (c *wellKnownReadyController) sync(ctx context.Context, controllerContext f
 	}
 
 	// the code below this point triggers status updates, unify status update handling in defer
-	statusUpdates := []v1helpers.UpdateStatusFunc{}
+	var available, progressing *applyoperatorv1.OperatorConditionApplyConfiguration
+	status := applyoperatorv1.OperatorStatus()
 	defer func() {
-		if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, statusUpdates...); updateErr != nil {
+		// only add valid conditions to the status. This is to prevent unintended changes in current behavior,
+		// where the sync function may return without setting a progressing condition.
+		conditions := []*applyoperatorv1.OperatorConditionApplyConfiguration{
+			available,
+			progressing,
+		}
+		for _, condition := range conditions {
+			if condition != nil {
```

Comment:
what happens to a condition that is `nil` ? I think it will be removed. Is this what we want :) ?

Reply:
IIUC it won't be removed because the field is a [listType=map](https://github.com/openshift/api/blob/b66574fab23f29af23c8ea2abd9641bae44b388e/operator/v1/types.go#L118-L119). Based on this, we should only update the condition we are specifically interested in and leave the other existing conditions untouched.

This logic (only apply valid, non-nil conditions) was introduced to align with the current behavior. For example, imagine get into this branch:

https://github.com/openshift/cluster-authentication-operator/pull/705/files#diff-b6f4bf160fd7e801eadfbfac60b84bd00fcad21a0979c2d730549f10db015645R136-L125

Under the current behavior, only the available condition is set, the progressing condition is ignored.

I did try to set the progressing condition unconditionally, but that's not what CI expects.

Does that make sense?

Reply:
with SSA and listType=map, the behavior goes liek this

1. SSA for fieldmanager/one that sets condition/foo and condition/bar
2. objectect has condition/foo and condition/bar
3. SSA  for fieldmanager/one, sets condition/bar
4. object has condition/bar, condition/foo is removed

I suspect we wnt to always set available adn progressing.

Reply:
hm, what if step 3 is:

3. SSA for fieldmanager/**two** sets condition/**baz**

then step 4 would be:

4. object has condition/bar, condition/foo **and** condition/baz?

is that correct?

Reply:
> is that correct?

Yes.  The fieldmanager/2 doesn't manipulate the conditions provided by fieldmanager/1

Reply:
@bertinatto are you willing to check the controls that have already been changed and make sure they don't have this issue?